### PR TITLE
Update negative-http2-interop-test-descriptions.md

### PR DESCRIPTION
**Pull request by ncteisen**
Small phrasing fix. More accurate now

### CODE REVIEW

File: doc/negative-http2-interop-test-descriptions.md [hunk 1]
```diff
@@ -49,7 +49,7 @@ server. The client should handle the goaway by switching to a new stream without
 the user application having to do a thing.
 
 Client Procedure:
- 1. Client sends two UnaryCall requests with:
+ 1. Client sends two UnaryCall requests (and sleeps for 1 second in-between).
```
Comment (adelez) [inline doc/negative-http2-interop-test-descriptions.md]: Can you comment on why this is needed? Is this to avoid the race condition?

Comment (ncteisen) [inline doc/negative-http2-interop-test-descriptions.md]: Yes, it is a quick fix for #9300, which will eventually need a better solution

Comment (adelez) [inline doc/negative-http2-interop-test-descriptions.md]: Can you please clarify with a TODO?

Comment (ncteisen) [inline doc/negative-http2-interop-test-descriptions.md]: Done

Comment (adelez) [inline doc/negative-http2-interop-test-descriptions.md]: LGTM.

**Pull request merged**